src/HMRServer.js

```diff
@@ -6,15 +6,63 @@ class HMRServer {
       this.wss = new WebSocket.Server({port: 0}, resolve);
     });
 
+    this.wss.on('connection', (ws) => {
+      if (this.unresolvedError) {
+        ws.send(JSON.stringify(this.unresolvedError))
+      }
+    });
+
     return this.wss._server.address().port;
   }
 
   stop() {
     this.wss.close();
   }
 
+  emitError(err) {
+    let message = typeof err === 'string' ? err : err.message;
```

Comment:
I mostly copied this code from [the logger](https://github.com/parcel-bundler/parcel/blob/ebadce69cd7ec003ad72a0c99d87675f4df50f7b/src/Logger.js#L46-L62) so that error messages/handling would be consistent. Perhaps it should be moved into a utility of some sort? Maybe `normalizeError(error, { colors: boolean }): { message, stack }`?

Reply:
Done in https://github.com/parcel-bundler/parcel/commit/67317ab5b4c618ed3492da236647a0a161d87562